test/smoke-decoders.js

```diff
@@ -0,0 +1,98 @@
+import _ from 'underscore';
+import chai from 'chai';
+import should from 'should';
+import * as es from 'event-stream';
+import {
+  fixture, bufferJs
+}
+from './fixture';
+import request from 'request';
+import CoreMock from './services/mock-core';
+import MockZKClient from './services/mock-zk';
+import {
+  EventEmitter
+}
+from 'events';
+import config from '../lib/config';
+import service from '../lib/service';
+import Shapefile from '../lib/decoders/shapefile';
+import KMZ from '../lib/decoders/kmz';
+import KML from '../lib/decoders/kml';
+import Disk from '../lib/decoders/disk';
+
+var res;
+var expect = chai.expect;
+
+function kmzDecoder() {
+  res = new EventEmitter()
+  return new KMZ(new Disk(res));
+}
+function shpDecoder() {
+  res = new EventEmitter()
+  return [new Shapefile(new Disk(res)), res]
+}
+function kmlDecoder() {
+  res = new EventEmitter()
+  return [new KML(new Disk(res)), res]
+}
+
+
+
+describe('decoder smoketest', () => {
```

Comment:
I think there should be a `describe` for each decoder, e.g. `describe('shapefile decoder')`. That way, you can have `describe`s and `it`s s.t. when concatenated, they read as a sentence, e.g. "shapefile decoder can read a multi-chunk shapefile". This is a Mocha convention and we use it for our mocha tests all over the frontend, so would be nice to keep that here.